### PR TITLE
Social embeds

### PR DESCRIPTION
**Pull request by dashaluna**

### CODE REVIEW

**Comment by dashaluna** [comment]
@mattheu not sure why FB URL didn't work.. if you have time could you have a look or can look together later on?

**Comment by dashaluna** [comment]
@mattheu so I have followed the youtube URL that wasn't replaced with video and [was output as string](http://youtu.be/8V7aA0RHK-Q) and it says it's private video.

What do we do in this case? Remove the URL from content or let it to be displayed as string?

**Comment by dashaluna** [comment]
@mattheu review & merge

**Comment by mattheu** [comment]
I think it would be neater to use `render_post_content` method that is already hooked in on `simple_fb_pre_render` rather than adding `render_social_embeds`?

**Comment by dashaluna** [comment]
`render_post_content` uses `dom` for post content manipulation. Probably better to rename it to `render_post_content_dom` and add another `post_content` filter that acts like a hook?

**Comment by mattheu** [comment]
No I think its fine - I just mean we should hook in on `simple_fb_pre_render` and `simple_fb_before_feed` once and add all our custom filters in the callback. `render_post_content` doesn't use DOM - but it adds a filter that does

**Comment by dashaluna** [comment]
@mattheu review & merge :)

File: simple-facebook-instant-articles.php [hunk 1]
```diff
@@ -240,6 +231,17 @@ public function image_shortcode( $atts, $content = '' ) {
...
+		// Shortcodes - overwrite WP native ones with FB IA format.
+		add_shortcode( 'gallery', array( $this, 'gallery_shortcode' ) );
+		add_shortcode( 'caption', array( $this, 'image_shortcode' ) );
+
+		// Render social embeds into FB IA format.
+		add_filter( 'embed_handler_html', array( $this, 'fb_formatted_social_embeds' ), 10, 3 );
+		add_filter( 'embed_oembed_html', array( $this, 'fb_formatted_social_embeds' ), 10, 4 );
+
+		// Render post content via DOM - to format it into FB IA format.
+		// DO it last, so content was altered via WP native hooks as much as possible.
```
Comment (mattheu) [inline simple-facebook-instant-articles.php:244]: You commented `DO it last` - but you're not hooking the filter in late. If you need this - I would hook in on `the_content` priority 100 or something. As the above code is affecting different filters - I don't think the order here makes any difference!

**Comment by dashaluna** [comment]
@mattheu "third" time lucky :) Review & merge

**Comment by mattheu** [comment]
:+1:

**Pull request merged**